### DISCUSSION TITLE
Custom input not working

### DISCUSSION THREAD

**Discussion by klavsbuss** [newcat/baklavajs#243] [Q&A] [answered]
@newcat can you please help me figuring out why my custom (`TextAreaOption`) input component is not working, even though it uses same listener as `InputOption` component. What am I doing wrong? [My Code](https://codesandbox.io/s/vue3-baklavajs-65zz10?file=/src/App.vue:1405-1416)
![image](https://user-images.githubusercontent.com/18184893/211220148-c71588f4-2973-443e-9273-030fdda6fd4d.png)

**Comment by newcat** [comment, marked as answer]
For Vue 3, adding the `emits` property is necessary. Also, you can remove the general listener stuff, since it is your custom component and you can add listeners if needed. This makes it more readable.

```vue
<template>
  <div>
    <textarea :value="value" @input="$emit('input', $event.target.value)" />
  </div>
</template>

<script>
export default {
  components: {},
  props: ["value", "name", "option", "node"],
  emits: ["input"],
};
</script>
```

**Reply by klavsbuss** [reply]
@newcat Works, you're the best! thanks
